src/psfmachine/machine.py

```diff
@@ -713,19 +818,36 @@ def build_time_model(self, plot=False):
         dy = dy.data * u.deg.to(u.arcsecond)
 
         A_c = _make_A_cartesian(
-            dx, dy, n_knots=self.n_time_knots, radius=self.time_radius
+            dx,
+            dy,
+            n_knots=self.n_time_knots,
+            radius=self.time_radius,
+            spacing=self.cartesian_knot_spacing,
         )
         A2 = sparse.vstack([A_c] * time_binned.shape[0], format="csr")
         # Cartesian spline with time dependence
-        A3 = sparse.hstack(
-            [
-                A2,
-                A2.multiply(time_binned.ravel()[:, None]),
-                A2.multiply(time_binned.ravel()[:, None] ** 2),
-                A2.multiply(time_binned.ravel()[:, None] ** 3),
-            ],
-            format="csr",
-        )
+        if hasattr(self, "pos_corr1") and self.time_corrector == "pos_corr":
+            # Cartesian spline with poscor dependence
+            A3 = sparse.hstack(
+                [
+                    A2,
+                    A2.multiply(poscorr1_binned.ravel()[:, None]),
+                    A2.multiply(poscorr2_binned.ravel()[:, None]),
+                    A2.multiply((poscorr1_binned * poscorr2_binned).ravel()[:, None]),
+                ],
+                format="csr",
+            )
+        else:
+            # Cartesian spline with time dependence
+            A3 = sparse.hstack(
+                [
+                    A2,
+                    A2.multiply(time_binned.ravel()[:, None]),
+                    A2.multiply(time_binned.ravel()[:, None] ** 2),
+                    A2.multiply(time_binned.ravel()[:, None] ** 3),
+                ],
+                format="csr",
+            )
```

Comment:
It seems like we need some helper functions to abstract out the matrix building

Reply:
this makes sense. I'll add a helper function in `utils.py` to take care of this